### PR TITLE
user made

### PR DESCRIPTION
**Pull request by ileuxali**
user entity made, still need to replace id generator with java builtin, let me know what other getter functions or any functions are required from user. added public profile class, working on it with hannah.

### CODE REVIEW

**Review by helenaglow — APPROVED** [review]
It looks good to me! There are changes to the initial design of user (i.e. changing variable names and organizing the class to ensure easier readability). It is bound to change as it is initial design, so for the meantime this is good!

File: src/main/java/Entities/User.java [hunk 1]
```diff
@@ -4,37 +4,64 @@
...
-    public User(String username, String password){
+    // maps username (key) to user ID (value) maybe not in here but somewhere else store
+    private static HashMap<Integer, String> UserIDMap;
+
+    public User(String username, String password, String name, String email, UserPublicProfile publicProile){
+        // will look into java builtin later
         Random rand = new Random();
-        Integer id = rand.nextInt();
-        while (UserIDMap.containsValue(id)){
+        Integer id = (Integer) rand.nextInt();
```
Comment (hannahpan1) [inline src/main/java/Entities/User.java:24]: Why do we do rand.nextInt() here? Could we not just assign it to (Integer) rand?

**Review by hannahpan1 — APPROVED** [review]

File: src/main/java/Entities/User.java [hunk 2]
```diff
@@ -4,37 +4,64 @@
...
-    private class UserPublicProfile {
-        private HashMap<String, String> preferences;
-        private UserPublicProfile(){
-//            this.preferences = HashMap<String, String> ;
-        }
-    }
+    private String email;
 
-    public User(String username, String password){
+    // maps username (key) to user ID (value) maybe not in here but somewhere else store
```
Comment (aaryavatsa) [inline src/main/java/Entities/User.java:18]: The line below initializes a HashMap<Integer, String>, so I think you just switched the order around in this doc comment.

**Review by aaryavatsa — APPROVED** [review]
Nicely done, I think your getters are comprehensive -- can't think of anything else that could be needed for my use case (but we can always add more if needed).

I think we could also discuss potentially making a UserFactory class later on to refactor some of this user creation code and make this cleaner. But perhaps it only seems this long because we haven't implemented the random id java builtin.

**Pull request merged**